File: abc/abc177/abc177_c.cs
// https://atcoder.jp/contests/abc177/submissions/16376028

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var mod = 1000000007;

        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split().Select(long.Parse).ToArray();

        var sum = 0L;
        for (int i = 0; i < N; i++)
        {
            sum += A[i];
            sum %= mod;
        }

        var result = 0L;
        for (int i = 0; i < N; i++)
        {
            sum -= A[i];
            if (sum < 0)
            {
                sum += mod;
            }
            result += A[i] * sum;
            result %= mod;
        }

        Console.WriteLine(result);
    }
}

File: abc/abc177/abc177_c.fs
// https://atcoder.jp/contests/abc177/submissions/16376925

let MOD = 1000000007L
let add x y = (x + y) % MOD
let sub x y = (x + MOD - y) % MOD

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int64

let mutable sum = 0L

for i in [ 0..N - 1 ] do
    sum <- add sum A.[i]

let mutable result = 0L

for i in [ 0..N - 1 ] do
    sum <- sub sum A.[i]
    if sum < 0L then sum <- add sum MOD
    result <- add result (A.[i] * sum)

result |> stdout.WriteLine
